design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+
+## Cluster Stacks
+
+Cluster Stacks can be thought of as a higher level of privilege than other types of stacks, similar to how kernel modules work in Linux.
+The controllers contained in a cluster stack will usually reconcile resources that are external to the cluster, for example cloud provider managed services.
+These controllers will also likely be written as full fledged controller-runtime based implementations, but that is not a requirement.
+
+Let's look at a specific example of the GCP cluster stack to understand the security and isolation model more thoroughly.
+First, the user will create an instance of a `ClusterStackInstall` that specifies which cluster stack they want to install.
+Then, when the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running.  This will serve as the runtime identity of the controller.
+* A `ClusterRoleBinding` is created at the cluster level to bind the `ClusterRole` and the namespaced `ServiceAccount`
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the GCP controllers.
+
+Because the stack is given a `ClusterRole` and `ClusterRoleBinding`, its permissions to the allowed resources are across all namespaces.
+
+### Architecture Diagram - Cluster Stack
+
+The diagram below summarizes the artifacts in the control plane after the GCP stack has been installed and a couple example `MySQLInstance` claims have been created.
+
+![Diagram of a Cluster Stack Installed](./images/cluster-stack-installed.png)
+
+### Environments
+
+While cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
+This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+
+#### Cluster Stack Single Instance Version Limitation
+
+One caveat of this approach is that because there is only a single instance of the cluster stack in the cluster, it is therefore shared across all environments.
+This means that all environments must use the same version of the stack.
```

Comment:
What if a user installs the same stack but with a different name in the install request?

Reply:
This case should encounter an error since this is not a valid case that we support.

Reply:
More for curiosity, but what is the expected error? Is it an error, or is the behavior undefined and not explicitly supported?

Reply:
I think it's a case that isn't explicitly handled right now so the behavior may be undefined currently.